Jenkinsfile

```diff
@@ -0,0 +1,82 @@
+pipeline {
+    agent none
+    environment {
+        projectName = 'DDF'
+    }
+    stages {
+        stage('Setup') {
+            steps{
+                notifySlack('good', "STARTED: ${JOB_NAME} ${BUILD_NUMBER} ${BUILD_URL}")
+            }
+        }
+        stage('Parallel Build') {
+            //TODO Parallel Stage Syntax may soon be changing for declaritive Jenkinsfiles - https://issues.jenkins-ci.org/browse/JENKINS-41334
+            steps{
+                parallel(
+                    linux: {
+                        node('linux-small') {
+                            timeout(time: 3, unit: 'HOURS') {
+                                checkout scm
+                                withMaven(maven: 'M3', mavenSettingsConfig: 'codice-maven-settings') {
+                                    sh 'mvn clean install'
+                                }
+                            }
+                        }
+                    }, windows: {
+                        node('proxmox-windows'){
+                            timeout(time: 3, unit: 'HOURS') {
+                                checkout scm
+                                withMaven(maven: 'M3', mavenSettingsConfig: 'codice-maven-settings') {
+                                    bat 'mvn clean install'
+                                }
+                            }
+                        }
+                    }
+                )
+            }
+        }
+        stage('Static Analysis') {
+            agent { label 'linux-small' }
+            steps{
+                timeout(time: 2, unit: 'MINUTES') {
+                    sh """echo Static Analysis run of "$projectName" """
+                    //TODO DDF-2946 Add SA tools
+                }
+            }
+            post {
+                failure {
+                    sh """Static Analysis failure"""
+                }
+            }
+        }
+        stage('Deploy') {
+            agent { label 'linux-small' }
+            //TODO Decide on conditional logic
+            when {
+                //TODO flow control for release/CICD
+                expression { environment.JOB_NAME == 'CICD' }
+            }
+            steps{
+                timeout(time: 2, unit: 'MINUTES') {
+                    sh """echo Deploy the artifacts of "$projectName" """
+                }
+            }
+        }
+    }
+    post {
+        success {
+            notifySlack('good', "SUCCESS: ${JOB_NAME} ${BUILD_NUMBER}")
```

Comment:
FYI, I started looking at how to put notifications into a shared library so that we can have a bit more flexibility. Things like having multiple places that notifications are sent to (slack, email, ...), being able to override the default slack channel and a few other things.